### PR TITLE
feat(diagnostic): Different signs for different types of diagnostics

### PR DESCRIPTION
**Pull request by ADIX7**
Add option to set different signs for different types of diagnostics.
The current default config matches the behaviour of the current default config.

***This is a breaking change in the config***

### CODE REVIEW

**Comment by ADIX7** [comment]
If this implementation works for you, I will update the readme too.

File: lua/satellite/handlers/diagnostic.lua [hunk 1]
```diff
@@ -59,10 +64,24 @@ function handler.setup(config0, update)
       buf_diags[bufnr] = args.data.diagnostics
 
       vim.schedule(update)
-    end
+    end,
   })
 end
 
+local function get_mark(severity, count)
+  local type = 'hint'
```
Comment (lewis6991) [inline lua/satellite/handlers/diagnostic.lua:72]: `type` is a builtin in Lua. Can you rename this to something else?

**Review by lewis6991 — APPROVED** [review]
LGTM:

Prefix the commit messages `feat(diagnostic)!: ...` so people update at least know there's a breaking change.

Alternatively we can respect the old config if there are integer keys in the table.

**Comment by ADIX7** [comment]
Fixed and chaged to `feat(diagnostic)!:...`

I would add backward compatibility, but I don't know hoe can I test for integer keys. I'm new to lua and neovim plugins.

**Comment by lewis6991** [comment]
Just do `if config[1] then -- old config`.

**Pull request merged**